newa/cli.py

```diff
@@ -511,13 +521,35 @@ def _jira_fake_id_generator() -> Generator[str, int, None]:
             # Use to prevent endless loop over the issue actions.
             endless_loop_check: dict[str, int] = {}
 
+            # updates
+            def _update_action_with_defaults(
+                    action: IssueAction,
+                    defaults: Optional[IssueAction] = None) -> IssueAction:
+                new_action = copy.deepcopy(action)
+                if not isinstance(defaults, IssueAction):
+                    return new_action
+                for attr_name in dir(defaults):
+                    attr = getattr(defaults, attr_name)
+                    if attr and (not attr_name.startswith('_') or callable(attr)):
+                        if attr_name == 'fields' and defaults.fields:
+                            if new_action.fields:
+                                new_action.fields.update(copy.deepcopy(defaults.fields))
+                            else:
+                                setattr(new_action, attr_name, copy.deepcopy(defaults.fields))
+                        elif not getattr(new_action, attr_name, None):
+                            setattr(new_action, attr_name, copy.deepcopy(attr))
+                return new_action
+
             # Iterate over issue actions. Take one, if it's not possible to finish it,
             # put it back at the end of the queue.
             while issue_actions:
                 action = issue_actions.pop(0)
 
                 ctx.logger.info(f"Processing {action.id}")
 
+                # update action object with default attributes when not present
+                action = _update_action_with_defaults(action, config.defaults)
```

Comment:
Why is handling of defaults done here in the loop and not during `config = IssueConfig.from_yaml_with_include` call?

Reply:
Well, I guess I see that function for loading YAML, not doing semantic evaluation. Are you strongly against keeping it here?

Reply:
> Well, I guess I see that function for loading YAML, not doing semantic evaluation.

That function already extends YAML loader with 'include'  and isn't the 'default' closer to syntax? From YAML perspective it is just a default dictionary to update with whatever it reads from the rest of the file.

IMO 'issue_actions' should contain "final" classes before they reach 'while' loop.

Reply:
I have pushed an update, pls review.

Reply:
I don't see 'resolve' button so please mark this thread as resolved.